Client.go

```diff
@@ -59,15 +169,40 @@ type successResponse struct {
 	Data interface{} `json:"data"`
 }
 
+// BaseURL return the base url of the rest API client
+func (c *Client) BaseURL() string {
```

Comment:
Are these functions required by the user, do we need these?

Reply:
I would prefer we didn't expose this and the client etc just used the internal config struct

Reply:
Yes, I find it convenient, that allow users to have a client config and then 'clone' to other client easily. Users can hold multiple clients connecting to different servers? Or multiple client to the same server with different username/password ... etc